File: cs/Garden/GardenId.cs
namespace Garden;

/// <summary>
/// Garden Id should be made of exactly 7 characters,
/// all digits from 1 to 7
/// </summary>
public class GardenId
{
    /// <summary>
    /// Check that the value is a valid Garden ID
    /// </summary>
    public static bool Check(string value)
    {
        if (value.Length != 7)
        {
            return false;
        }

        foreach (var c in value)
        {
            if (!char.IsDigit(c))
            {
                return false;
            }
        }

        return true;
    }
}

File: java/src/main/java/fr/arolla/garden/GardenId.java
package fr.arolla.garden;

public class GardenId {
  public static boolean check(String value) {
    if (value.length() != 7) {
      return false;
    }

    for (var c : value.toCharArray()) {
      if (!Character.isDigit(c)) {
        return false;
      }
    }

    return true;
  }
}
